src/workspaceViewer.ts

```diff
@@ -84,24 +88,33 @@ export class WorkspaceItem extends TreeItem {
 		str: string,
 		type: string,
 		length: number,
-		collapsibleState: TreeItemCollapsibleState
+		collapsibleState: TreeItemCollapsibleState,
+		dim?: number[]
 	) {
 		super(label, collapsibleState);
-		this.description = str;
+		this.description = this.getDescription(dim, str);
 		this.tooltip = `${label} (${rClass}, length of ${length})`;
 		this.contextValue = type;
 	}
+
+	private getDescription(dim: number[], str: string): string {
+		if (dim !== undefined) {
+			return `${dim[0]} obs. of ${dim[1]} variables`
```

Comment:
This should apply to `matrix` and `array` (which have `dim`) but somehow it does not as I tested.

Reply:
Just checked `init.R` and we only write `dim` when the object is `list` or `environment` which has `names()`. In this case, only `data.frame` will have `dim`.

Reply:
Oops, yes that's my bust, I completely forgot about non-data.frame objects. Thanks for the heads up!